### PR TITLE
Fix reward dataloader

### PR DESCRIPTION
**Pull request by tcbegley**
Use dataloaders + tensorclasses to the reward training pipeline

### CODE REVIEW

File: trainers/reward_trainer.py [hunk 1]
```diff
@@ -345,3 +363,20 @@ def train(self):
...
+        for split in ["train", "val"]:
+            losses = torch.zeros(self.eval_iters)
+            for k in range(self.eval_iters):
+                batch = self.get_batch(split)
+                with ctx:
+                    logits, loss = model(batch.prompt, batch.reward)
+                losses[k] = loss.item()
+            out[split] = losses.mean()
+        model.train()
+        return out
```
Comment (tcbegley) [inline trainers/reward_trainer.py]: #2 should render this obsolete
